src/aui-widget/tests/unit/js/aui-widget-position-align-suggestion-tests.js

```diff
@@ -0,0 +1,423 @@
+YUI.add('aui-widget-position-align-suggestion-tests', function(Y) {    
+    var suite = new Y.Test.Suite('aui-widget-position-align-suggestion-tests');
+    var TestWidget =  Y.Base.create('testwidget', Y.Widget, [
+            Y.WidgetPosition,
+            Y.WidgetStdMod,
+            Y.WidgetPositionAlign,
+            Y.WidgetPositionAlignSuggestion,
+            Y.WidgetPositionConstrain,
+            Y.WidgetTrigger
+        ], {
+            initializer: function() {
+                this.get('boundingBox').addClass('popover').append('<div> Hei! </div>');
+                Y.on('windowresize', Y.bind(this._onResize, this));
+            },
+
+            _onResize: function() {
+                this.suggestAlignment(Y.one('#bottom'));
+            }
+        });
+
+    suite.add(new Y.Test.Case({
+
+        name: 'AUI WidgetPositionAlignSuggestion Unit Tests',
+
+        init: function() {
+            this._buttom = Y.one('#buttom');
+            this._secondButton = Y.one('#buttomTwo');
+            this._constrainElement = Y.one('#containerConstrain');
+            this._auxiliarButtom = Y.Node.create('<button>Button Target</button>');
+
+            this._setTargetsDefaultPosition();
+        },
+
+        setUp: function(){
+            this._setTargetsDefaultPosition();
+        },
+
+        destroy: function() {
+            this._buttom.remove();
+            this._secondButton.remove();
+            this._auxiliarButtom.remove();
+        },
+
+        tearDown: function() {
+            if (this._widget) {
+                this._widget.destroy();
+            }
+        },
+
+        createWidget: function(config) {
+            this._widget = new TestWidget(Y.merge({
+                align: {
+                    node: '#buttom'
+                },
+                constrain: true
+            }, config || {})).render();
+        },
+
+        _setTargetsDefaultPosition: function() {
+            this._buttom.setStyle('left', '50px');
+            this._buttom.setStyle('right', 'auto');
+            this._buttom.setStyle('bottom', 'auto');
+            this._buttom.setStyle('top', '451px');
+
+            this._secondButton.setStyle('top', '100px');
+            this._secondButton.setStyle('left', '150px');
+            this._secondButton.setStyle('bottom', 'auto');
+            this._secondButton.setStyle('right', 'auto');
+        },
+
+        'should find best position even if its target is setted late': function() {
+            this.createWidget({
+                align: undefined,
+                position: 'bottom'
+            });
+
+            this._widget.set('align', {
+                node: this._buttom
+            });
+
+            this.wait(function() {
+                Y.Assert.isTrue(Y.one('.popover').inViewportRegion());
+            }, 100);
+        },
+
+        'shouldn`t find the best position if the constrain is false': function() {
```

Comment:
This should be "shouldn\'t" instead of "shouldn`t", since that's the pattern for tests docs.